packages/forms-web-app/src/dynamic-forms/question.js

```diff
@@ -102,71 +113,167 @@ class Question {
+	 * @param {ExpressRequest} req
+	 * @param {JourneyResponse} journeyResponse - current journey response, modified with the new answers
+	 * @returns {Promise.<Object>}
+	 */
+	async getDataToSave(req, journeyResponse) {
+		// set answer on response
+		let responseToSave = { answers: {} };
+
+		responseToSave.answers[this.fieldName] = req.body[this.fieldName];
+
+		for (const propName in req.body) {
+			if (propName.startsWith(this.fieldName + '_')) {
+				responseToSave.answers[propName] = req.body[propName];
+			}
 		}
 
-		return res.render(`dynamic-components/${this.viewFolder}/index`, {
-			...viewModel,
-			...customViewData
-		});
-	};
+		journeyResponse.answers[this.fieldName] = responseToSave[this.fieldName];
+
+		return responseToSave;
+	}
+
+	/**
+	 * check for validation errors
+	 * @param {ExpressRequest} req
+	 * @param {Journey} journey
+	 * @param {Section} sectionObj
+	 * @returns {Object|undefined} returns the view model for displaying the error or undefined if there are no errors
+	 */
+	checkForValidationErrors(req, sectionObj, journey) {
+		const { body } = req;
+		const { errors = {}, errorSummary = [] } = body;
+		if (Object.keys(errors).length > 0) {
+			return this.prepQuestionForRendering(sectionObj, journey, {
+				errors,
+				errorSummary
+			});
+		}
+
+		return;
+	}
+
+	/**
+	 * check for errors after saving, by default this does nothing
+	 * @param {ExpressRequest} req
+	 * @param {Journey} journey
+	 * @param {Section} sectionObj
+	 * @returns {Object|undefined} returns the view model for displaying the error or undefined if there are no errors
+	 */ //eslint-disable-next-line no-unused-vars
+	checkForSavingErrors(req, sectionObj, journey) {
+		return;
+	}
+
+	/**
+	 * returns answer to the question formatted for a list view
+	 * @param {Object} answer
+	 * @returns {string}
+	 */
+	formatAnswerForSummary(answer) {
+		return this.altText ?? answer ?? 'Not started';
+	}
+
+	/**
+	 * @param {JourneyResponse} journeyResponse
+	 * @param {Promise<void>} responseToSave
+	 */
+	async saveResponseToDB(journeyResponse, responseToSave) {
+		const encodedReferenceId = encodeURIComponent(journeyResponse.referenceId);
+		await patchQuestionResponse(journeyResponse.journeyId, encodedReferenceId, responseToSave);
+	}
+
+	/**
+	 * Handles redirect after saving
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {string} sectionSegment
+	 * @param {string} questionSegment
+	 * @returns {void}
+	 */
+	handleNextQuestion(res, journey, sectionSegment, questionSegment) {
+		return res.redirect(journey.getNextQuestionUrl(sectionSegment, questionSegment, false));
+	}
+
+	/**
+	 * Save the answer to the question
+	 * @param {ExpressRequest} req
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {Section} section
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns {Promise<void>}
+	 */
+	async saveAction(req, res, journey, section, journeyResponse) {
```

Comment:
I've created a common saveAction
multi file uploads check for validation errors after saving, the others do not use checkForSavingErrors

An alternative I've been considering is potentially have an array of functions to pipe together

can then control exactly what order to call these functions and can override some if necessary, could help reduce the duplication + minor variations in the add more override of this